booktown.md

```diff
@@ -0,0 +1,42 @@
+booktown=# \d
+booktown=# SELECT * FROM title;
+ERROR:  relation "title" does not exist
+LINE 1: SELECT * FROM title;
+                      ^
+LINE 1: SELECT * FROM books WHERE name = 'Little Women';
+
+booktown=# SELECT * FROM books WHERE title = 'Python';
+
+
+booktown=# \d subjects
+
+Indexes:
+    "subjects_pkey" PRIMARY KEY, btree (id)
+
+booktown-# SELECT * FROM subjects WHERE location = 'Main St';
+
+booktown=# SELECT * FROM subjects WHERE location = 'Main St';
+
+ERROR:  relation "subject" does not exist
+LINE 1: SELECT * FROM subject ORDER BY location DESC\
+
+Indexes:
+    "books_id_pkey" PRIMARY KEY, btree (id)
+    "books_title_idx" btree (title)
+
+booktown=# SELECT * FROM books;
+
+
+booktown=# SELECT books.title FROM books 
+booktown-# ;
+
+booktown=# SELECT books.title FROM books
+booktown-# INNER JOIN subjects on books.subjects_id = subjects.id;
+ERROR:  column books.subjects_id does not exist
+LINE 2: INNER JOIN subjects on books.subjects_id = subjects.id;
+                               ^
+HINT:  Perhaps you meant to reference the column "books.subject_id".
+booktown=# SELECT books.title FROM books                                                                                                                                                                    INNER JOIN subjects on books.subject_id = subjects.id;
+            title            
+
+
```

Comment:
Here are solutions for this -
Order:
1. select * from subjects order by subject;
2. select * from subjects order by location;

Where:
1. select * from books where title = 'Little Women';
2. select * from books where title like '%Python%';
3. select * from subjects where location = 'Main St' order by subject;

Joins: 
1. select books.title from books join subjects on books.subject_id = subjects.id where subjects.subject = 'Computers';
2. select books.title, authors.first_name, authors.last_name, subjects.subject from books join authors on books.author_id = authors.id join subjects on books.subject_id = subjects.id;
3. select books.title, stock.retail from stock join editions on stock.isbn = editions.isbn join books on editions.book_id = books.id order by stock.retail desc;
4. select books.title, editions.isbn, publishers.name, stock.retail from books join editions on books.id = editions.book_id join stock on editions.isbn = stock.isbn join publishers on editions.publisher_id = publishers.id where books.title = 'Dune';
5. select customers.first_name, customers.last_name, shipments.ship_date, books.title from shipments join editions on shipments.isbn =editions.isbn join customers on shipments.customer_id = customers.id join books on editions.book_id = books.id order by shipments.ship_date;

Grouping and Counting:
1. select count(*) from books;
2. select count(*) from (select distinct location from subjects) as stuff;
3. select location, count(location) from subjects;
4. select books.id, books.title, count(editions.book_id) from books join editions on books.id = editions.book_id group by books.id;